Source/WebKit/UIProcess/Automation/WebDriverBidiProcessor.cpp

```diff
@@ -111,19 +117,183 @@ void WebDriverBidiProcessor::sendMessageToFrontend(const String& message)
 }
 
 
-// MARK: Inspector::BrowsingContextDispatcherHandler methods.
+// MARK: Inspector::BidiBrowsingContextDispatcherHandler methods.
 
-void WebDriverBidiProcessor::navigate(const BrowsingContext& browsingContext, const String& url, std::optional<Inspector::Protocol::BidiBrowsingContext::ReadinessState>&&, CommandCallbackOf<String, String>&& callback)
+void WebDriverBidiProcessor::activate(const BrowsingContext& browsingContext, CommandCallback<void>&& callback)
 {
     RefPtr session = m_session.get();
     if (!session)
         ASYNC_FAIL_WITH_PREDEFINED_ERROR(InternalError);
 
-    ASYNC_FAIL_WITH_PREDEFINED_ERROR(NotImplemented);
+    RefPtr webPageProxy = session->webPageProxyForHandle(browsingContext);
+    if (!webPageProxy)
+        ASYNC_FAIL_WITH_PREDEFINED_ERROR(WindowNotFound);
+
+    // FIXME: detect non-top level browsing contexts, returning `invalid argument`.
+
+    session->switchToBrowsingContext(browsingContext, emptyString(), [callback = WTFMove(callback)](CommandResult<void>&& result) {
+        if (!result) {
+            callback(makeUnexpected(result.error()));
+            return;
+        }
+
+        callback({ });
+    });
+}
+
+void WebDriverBidiProcessor::close(const BrowsingContext& browsingContext, std::optional<bool>&& optionalPromptUnload, CommandCallback<void>&& callback)
+{
+    RefPtr session = m_session.get();
+    if (!session)
+        ASYNC_FAIL_WITH_PREDEFINED_ERROR(InternalError);
+
+    // FIXME: implement `promptUnload` option.
```

Comment:
if we're not using `optionalPromptUnload` right now, is there a reason we're handling it?  id much prefer not having the parameter at all (both here and in the protocol `.json`) if it's not supported

ditto with many other FIXME below (and to be clear i have no issue with having a FIXME comment, just that we also shouldn't support the code if it's not implemented)